dirty_cat/datetime_encoder.py

```diff
@@ -2,50 +2,70 @@
 import pandas as pd
 
 from warnings import warn
-from typing import List
+from typing import List, Literal, Dict, Optional
 
 from sklearn import __version__ as sklearn_version
 from sklearn.base import BaseEstimator, TransformerMixin
 from sklearn.utils.validation import check_is_fitted
 
 from dirty_cat.utils import check_input, Version
 
-# Some functions need aliases
-WORD_TO_ALIAS = {"year": "Y", "month": "M", "day": "D", "hour": "H", "minute": "min", "second": "S",
-                 "millisecond": "ms", "microsecond": "us", "nanosecond": "N"}
-TIME_LEVELS = ["year", "month", "day", "hour", "minute", "second", "millisecond", "microsecond", "nanosecond"]
+
+WORD_TO_ALIAS: Dict[str, str] = {
+    "year": "Y", "month": "M", "day": "D",
+    "hour": "H", "minute": "min", "second": "S",
+    "millisecond": "ms", "microsecond": "us", "nanosecond": "N"
+}
+TIME_LEVELS: List[str] = list(WORD_TO_ALIAS.keys())
+AcceptedTimeValues = Literal["year", "month", "day",
+                             "hour", "minute", "second",
+                             "millisecond", "microsecond", "nanosecond"]
 
 
 class DatetimeEncoder(BaseEstimator, TransformerMixin):
     """
-    This encoder transforms each datetime column into several numeric columns corresponding to temporal features,
-    e.g year, month, day...
-    Constant extracted features are dropped ; for instance, if the year is always the same in a feature, the extracted "year" column won't be added.
-    If the dates are timezone aware, all the features extracted will correspond to the provided timezone.
+    This encoder transforms each datetime column into several numeric columns
+    corresponding to temporal features, e.g year, month, day...
+    Constant extracted features are dropped; for instance, if the year is
+    always the same in a feature, the extracted "year" column won't be added.
+    If the dates are timezone aware, all the features extracted will correspond
+    to the provided timezone.
 
     Parameters
     ----------
-    extract_until : {"year", "month", "day", "hour", "minute", "second", "millisecond", "microsecond", "nanosecond"}, default="hour"
-        Extract up to this granularity. If all features have not been extracted, add the "total_time" feature, which contains
-        the time to epoch (in seconds).
-        For instance, if you specify "day", only "year", "month", "day" and "total_time" features will be created.
+    extract_until : {"year", "month", "day", "hour", "minute", "second",
+                    "millisecond", "microsecond", "nanosecond"}, default="hour"
```

Comment:
Did you check that this renders well on the generated doc?

Reply:
Good catch, I've also noticed class variables are added in the generated doc, which is ugly.